node-repository/src/test/java/com/yahoo/vespa/hosted/provision/provisioning/AclProvisioningTest.java

```diff
@@ -36,12 +33,7 @@
  */
 public class AclProvisioningTest {
 
-    private ProvisioningTester tester;
-
-    @Before
-    public void before() {
-        this.tester = new ProvisioningTester(Zone.defaultZone(), createConfig());
-    }
+    private ProvisioningTester tester = new ProvisioningTester.Builder().build();
```

Comment:
The idea here is to throw away state before each test. Reusing it may work in some cases, but sharing state across tests is rarely a good idea. This applies to all tests that use this pattern.

Reply:
Yes, JUnit will by default create new instance of the test class for every test, see https://junit.org/junit5/docs/current/user-guide/#writing-tests-test-instance-lifecycle (Although this is junit 5 docs, the same behavior is on 4 too):
```
public class Test {
    private int a = 6;
    private List<String> b = new ArrayList<>();

    @Test
    public void test1() {
        System.out.println("test1 " + a + " " + b);
        a = 12252;
        b.add("test");
        System.out.println("test1 " + a + " " + b);
    }

    @Test
    public void abc123() {
        System.out.println("abc123 " + a + " " + b);
        a = 54;
        b.add("new");
        b.add("abc");
        System.out.println("abc123 " + a + " " + b);
    }

    @Test
    public void def456() {
        System.out.println("def456 " + a + " " + b);
        a = 4234;
        b = new LinkedList<>();
        System.out.println("def456 " + a + " " + b);
    }
}
```
Gives
```
abc123 6 []
abc123 54 [new, abc]
def456 6 []
def456 4234 []
test1 6 []
test1 12252 [test]
```

Reply:
Nice, didn't know that. 👍